fe/fe-core/src/main/java/com/starrocks/load/InsertOverwriteJobRunner.java

```diff
@@ -429,9 +445,18 @@ private void doCommit(boolean isReplay) {
             // try exception to release write lock finally
             final OlapTable targetTable = checkAndGetTable(db, tableId);
             tmpTargetTable = targetTable;
-            List<String> sourcePartitionNames = job.getSourcePartitionIds().stream()
-                    .map(partitionId -> targetTable.getPartition(partitionId).getName())
-                    .collect(Collectors.toList());
+            List<String> sourcePartitionNames = job.getSourcePartitionNames();
```

Comment:
This will still lead to behavioral changes. Is there any intention in the original design? @Astralidea

Reply:
I'm not entirely sure, but I feel like it should be fine.